src/simsopt/field/magneticfieldclasses.py

```diff
@@ -359,14 +362,44 @@ def __init__(self, r0=0.1, center=[0, 0, 0], I=5e5/np.pi, normal=[0, 0]):
         self.Inorm = I*4e-7
         self.center = center
         self.normal = normal
-        if len(normal) == 2:
-            theta = normal[0]
-            phi = normal[1]
-        else:
-            theta = np.arctan2(normal[1], normal[0])
-            phi = np.arctan2(np.sqrt(normal[0]**2+normal[1]**2), normal[2])
 
-        self.rotMatrix = np.array([
+        super().__init__(x0=self.get_dofs(), names=self._make_names(),external_dof_setter=CircularCoil.set_dofs_impl)
```

Comment:
Why is `external_dof_setter` needed here?  Typically, I've seen this used when there are dofs in `simsoptpp` that need to be set.  For example, when a Python class does not have dofs located in a cpp counterpart, we've simply had a `set_dofs` function Python-side like below:

https://github.com/hiddenSymmetries/simsopt/blob/4da52143991dd463d1a2e3bbd13a820dc73be338/src/simsopt/geo/surfacexyztensorfourier.py#L87

Reply:
Honestly, I am not 100% sure, but if I don't use the external_dof_setter and simply use the set_dofs, then the dofs are defined and updated, but not the class attributes, something that was confusing to me. For example, after declaring a CircularCoil object OBJ, if an optimizer changes the dofs, then OBJ.x shows the correct values, but OBJ.get_dofs() does not show the correct updated values (rather the initial values). Not sure how else to make this work otherwise, but with the external_dof_setter, then everything works fine.